### PR TITLE
in this commit is the full test for endpoing get/api/article/articlei…

### PR DESCRIPTION
**Pull request by rachelkwok00**
…d/comment

### CODE REVIEW

File: __tests__/endpoint.test.js [hunk 1]
```diff
@@ -174,4 +174,56 @@ describe('/api/articles', () => {
...
+                responseObj.map(obj=>{
+                  for (const key in obj) {
+  
+                  expect(obj).toBeInstanceOf(Object);
+                  expect(obj).toHaveProperty("comment_id", expect.any(Number));
+                  expect(obj).toHaveProperty("body", expect.any(String));
+                  expect(obj).toHaveProperty("article_id", expect.any(Number));
+                  expect(obj).toHaveProperty("votes", expect.any(Number));
+                  expect(obj).toHaveProperty("created_at", expect.any(String));
+                  }
```
Comment (NRMSMITH) [inline __tests__/endpoint.test.js]: it should also have the author of the comment here too - please make sure to check for that
- also, consider if you need this to be a map - which creates a new array - is there another array method that you could use to check what is inside each element?

Comment (NRMSMITH) [reply __tests__/endpoint.test.js]: - another point, do you need the for in loop?

File: __tests__/endpoint.test.js [hunk 2]
```diff
@@ -174,4 +174,56 @@ describe('/api/articles', () => {
...
+            });
+        });
+        test("Returned array should be sorted by ascending order of comments created at", () => {
+          return request(app).get("/api/articles/5/comments").expect(200).then(result => {
+  
+              const commentArr = result.body.comment;
+            
+              expect(commentArr).toBeSorted({key: "created_at", ascending: true})
+             
+          })
```
Comment (NRMSMITH) [inline __tests__/endpoint.test.js]: You're missing a 200 test - when a valid id is passed but there are no comments on that article :)

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Please have a look at the comments I've made for your tests -you'll have to figure out a way to get the additional test to work alongside your error handling :)

File: __tests__/endpoint.test.js (hunk 2 above)
Comment (rachelkwok00) [reply __tests__/endpoint.test.js]: i dont understand this

File: __tests__/endpoint.test.js [hunk 3]
```diff
@@ -4,174 +4,270 @@ const app = require('../app.js');
...
+      username: "butter_bridge",
+      body: "new comment"
+    };
+
+    return request(app).post("/api/articles/1/comments").send(newComment).expect(201).then(result => {
+
+      expect(result.body).toMatchObject({
+        author: "butter_bridge",
+        body: "new comment"
+      })
```
Comment (pippaamy) [inline __tests__/endpoint.test.js:250]: can we have another 201 that ignores unnecessary properties

File: __tests__/endpoint.test.js [hunk 4]
```diff
@@ -4,174 +4,270 @@ const app = require('../app.js');
...
+
+      });
+  });
+  test('GET:400 responds with an appropriate error message when given an invalid id', () => {
+    return request(app)
+      .get("/api/articles/not-a-id/comments")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe('Bad request');
+      });
```
Comment (pippaamy) [inline __tests__/endpoint.test.js:324]: what about if the username doesnt exist?

File: __tests__/endpoint.test.js [hunk 5]
```diff
@@ -4,174 +4,270 @@ const app = require('../app.js');
...
+  });
+
```
Comment (pippaamy) [inline __tests__/endpoint.test.js:326]: what about if they are missing the username or body?

**Review by pippaamy — COMMENTED** [review]
good job, a couple of tests to add- can you pop those in and pop in another pr ! well done :)

File: controllers/controller.js [hunk 6]
```diff
@@ -50,4 +54,53 @@ const {
...
+
+    const { article_id } = req.params;
+
+    Promise.all([
+      selectArticleById(article_id),
+      selectArticleComment(article_id)
+    ])
+
+    .then(([articleExist ,comment]) => {
+      if(!articleExist){
```
Comment (RichardNeat) [inline controllers/controller.js]: Do we need this error handler here? As we're invoking selectArticleById with our id that model should throw an error if that article doesn't exist based on previous logic. Try removing this and see if everything still works.

As a general rule we shouldn't have any logic implemented in the controller.

File: controllers/controller.js [hunk 7]
```diff
@@ -50,4 +54,53 @@ const {
...
+      })
+  };
+
+  exports.postComment = (req, res, next) => {
+    
+    const { article_id } = req.params;
+    const newComment  = req.body;
+
+  
+     Promise.all([
```
Comment (RichardNeat) [inline controllers/controller.js:84]: We don't need to worry about checking if the article exists here with a separate model call, if you try to insert a comment on an article that doesn't exist you will get a psql error :)

**Review by RichardNeat — APPROVED** [review]
Nice one Rachel! Really good stuff. I've made a couple of comments below for suggested changes, have a look and please drop me a message if you'd like anything clarifying :)

**Pull request merged**